UI/window-projector.cpp

```diff
@@ -848,8 +848,51 @@ void OBSProjector::mousePressEvent(QMouseEvent *event)
 					SLOT(OpenWindowedProjector()));
 
 		} else if (!this->isMaximized()) {
+
+			auto resizeToCustomScaleAction = [this](double scale) {
+				ResizeToScale(scale);
+			};
+
+			auto resizeToScaleAction = [this](QAction *action) {
+				double scale = action->property("scale_amount")
+						       .toDouble();
+				ResizeToScale(scale);
+			};
+
 			popup.addAction(QTStr("ResizeProjectorWindowToContent"),
 					this, SLOT(ResizeToContent()));
+			QMenu *scaleMenu = new QMenu(
+				QTStr("ResizeProjectorWindowToSourceScale"));
+
+			double scalePresets[] = {0.5, 1, 1.5, 2, 2.5, 3, 4};
+
+			for (size_t i = 0;
+			     i < sizeof(scalePresets) / sizeof(double); i++) {
+				QAction *scale = new QAction(
+					QString("%1x").arg(scalePresets[i]),
+					this);
+				scale->setProperty("scale_amount",
+						   scalePresets[i]);
+				connect(scale, &QAction::triggered,
+					std::bind(resizeToScaleAction, scale));
+				scaleMenu->addAction(scale);
+			}
+			QDoubleSpinBox *customScale = new QDoubleSpinBox();
+			customScale->setMinimum(0.1);
```

Comment:
```suggestion
			customScale->setMinimum(0.5);
```
The downside of 0.1 is that if you use arrow keys/scroll down to 0.1, then back up, you're stuck at x+0.1 until manually modified. By setting the minimum to 0.5, you'll never run into that issue.

Reply:
I did notice this, but unfortunately it prevents users from entering values of `0.125` and `0.25`

Reply:
What if we set the easy increment to 0.1 instead of 0.5 then? I personally expect people to use the presets significantly more often than the input.